qdrant-landing/content/documentation/guides/distributed_deployment.md

```diff
@@ -230,6 +235,153 @@ DELETE /cluster/peer/{peer_id}
+    # ... other collection parameters
+)
+```
+
+```typescript
+import { QdrantClient } from "@qdrant/js-client-rest";
+
+const client = new QdrantClient({ host: "localhost", port: 6333 });
+
+client.createCollection("{collection_name}", {
+    shard_number: 1,
+    sharding_method: "custom",
+    // ... other collection parameters
+});
+```
+
+```rust
+
+use qdrant_client::{
+    client::QdrantClient,
+    qdrant::{CreateCollection, ShardingMethod},
+};
+
+let client = QdrantClient::from_url("http://localhost:6334").build()?;
+
+client
+    .create_collection(&CreateCollection {
+        collection_name: "{collection_name}".to_string(),
+        shard_number: Some(1),
+        sharding_method: Some(ShardingMethod::Custom),
+        ..Default::default()
+    })
+    .await?;
+```
+
+In this mode, the `shard_number` means the number of shards per shard key, where points will be distributed evenly. For example, if you have 10 shard keys and a collection config with these settings:
+```
+{
+    "shard_number": 1,
+    "sharding_method": "custom"
+    "replication_factor": 2
+}
+```
+
+Then you will have `1 * 10 * 2 = 20` total physical shards in the collection.
+
+To specify the shard for each point, you need to provide the `shard_key` field in the point payload:
+
+```http
+PUT /collections/{collection_name}/points
+{
+    "points": [
+        {
+            "id": 1111,
+            "vector": [0.1, 0.2, 0.3]
+        },
+    ]
+    "shard_key": "user_1"
+}
+```
+
+```python
+from qdrant_client import QdrantClient
+from qdrant_client.http import models
+
+client = QdrantClient("localhost", port=6333)
+
+client.upsert_points(
+    collection_name="{collection_name}",
+    points=[
+        models.Point(
+            id=1111,
+            vector=[0.1, 0.2, 0.3],
+        ),
+    ],
+    shard_key="user_1",
+)
+```
+
+```typescript
+
+client.upsertPoints("{collection_name}", {
+    points: [
+        {
+            id: 1111,
+            vector: [0.1, 0.2, 0.3],
+        },
+    ],
+    shard_key: "user_1",
+});
+```
+
+```rust
+
+use qdrant_client::qdrant::{PointStruct, WriteOrdering, WriteOrderingType};
+
+client
+    .upsert_points_blocking(
+        "{collection_name}",
+        Some(vec![shard_key::Key::String("user_1".to_string())]),
+        vec![
+            PointStruct::new(
+                1111,
+                vec![0.1, 0.2, 0.3],
+                Default::default(),
+            ),
+        ],
+        None,
+    )
+    .await?;
+```
+
+<aside role="alert">
+When using custom sharding, IDs are only enforced to be unique within a shard key. This means that you can have multiple points with the same ID, if they have different shard keys.
+
+This is a limitation of the current implementation, and is an anti-pattern that should be avoided because it can create scenarios of points with the same ID to have different contents. In the future, we plan to add a global ID uniqueness check.
+</aside>
```

Comment:
Not sure about this one.

Maybe we should just state:

> Using the same point ID across multiple shard keys is not supported and should be avoided.

Reply:
Added your concise description first 👍